File: legacy/csharp/components/Strata.Domain/UnitOfWork/OptimisticLockException.cs
//  ##########################################################################
//  # File Name: OptimisticLockException.cs
//  # Copyright: 2011-2018, Sapientia Systems, LLC.
//  ##########################################################################

using System;

namespace Strata.Domain.UnitOfWork
{
    //////////////////////////////////////////////////////////////////////////
    /// <summary>
    /// Exception that indicates optimistic lock contention for an entity.
    /// </summary>
    ///  
    public
    class OptimisticLockException:
        CommitFailedException
    {
        public
        OptimisticLockException(String message):
            base(message) {}

        public
        OptimisticLockException(String message,Exception cause):
            base(message,cause) {}
    }
}

//  ##########################################################################

File: java/Components/Strata.Domain.Core/src/strata/domain/core/unitofwork/OptimisticLockException.java
//////////////////////////////////////////////////////////////////////////////
// OptimisticLockException.java
//////////////////////////////////////////////////////////////////////////////

package strata.domain.core.unitofwork;

public
class OptimisticLockException
    extends CommitFailedException
{
    public
    OptimisticLockException(String message)
    {
        super(message);
    }

    public
    OptimisticLockException(String message,Throwable cause)
    {
        super(message,cause);
    }
}

//////////////////////////////////////////////////////////////////////////////
